src/bsoncxx/util/functor.hpp

```diff
@@ -22,9 +22,89 @@ namespace bsoncxx {
 BSONCXX_INLINE_NAMESPACE_BEGIN
 namespace util {
 
+// TODO(MSVC): VS2015U1 Completely falls apart trying to honor the
```

Comment:
specifically, is_convertible returns the wrong results for conversions to std::function